drivers/crypto/crypto_stm32.c

```diff
@@ -45,7 +45,11 @@ LOG_MODULE_REGISTER(crypto_stm32);
 #define STM32_CRYPTO_TYPEDEF            AES_TypeDef
 #endif
 
-struct crypto_stm32_session crypto_stm32_sessions[CRYPTO_MAX_SESSION];
+#define USE_HAL_GENERIC                                                                            \
+	(DT_HAS_COMPAT_STATUS_OKAY(st_stm32_cryp) ||                                               \
+	 (DT_HAS_COMPAT_STATUS_OKAY(st_stm32_aes) && !DT_HAS_COMPAT_STATUS_OKAY(st_stm32l4_aes)))
+
+#define USE_HAL_L4 (DT_HAS_COMPAT_STATUS_OKAY(st_stm32l4_aes))
```

Comment:
Those 2 defines are not required. Checking `DT_HAS_COMPAT_STATUS_OKAY(st_stm32l4_aes)` should be sufficient.